scripts/dts/gen_defines.py

```diff
@@ -969,181 +292,211 @@ def encode_zephyr_multi_level_irq(irq, irq_num):
             irq_ctrl = irq_ctrl.interrupts[0].controller
         return irq_num
 
-    out_comment("IRQ macros from the 'interrupts' property",
-                blank_before=False)
-    for_each_ident(node, write_irqs_for_ident)
+    idx2val = {}
+    name2val = {}
+    for i, irq in enumerate(node.interrupts):
+        for cell_name, cell_value in irq.data.items():
+            cell_ident = str2ident(cell_name)
+
+            if cell_name == "irq":
+                if "arm,gic" in irq.controller.compats:
+                    cell_value = map_arm_gic_irq_type(irq, cell_value)
+                cell_value = encode_zephyr_multi_level_irq(irq, cell_value)
+
+            idx_macro = f"{node.z_path_id}_IRQ_IDX_{i}_CELL_{cell_ident}"
+            idx2val[idx_macro] = cell_value
+            if irq.name:
+                name2val[f"{node.z_path_id}_"
+                         f"IRQ_NAME_{str2ident(irq.name)}_"
+                         f"CELL_{cell_ident}"] = f"DT_{idx_macro}"
+
+    for macro, val in idx2val.items():
+        out_dt_define(macro, val)
+    for macro, val in name2val.items():
+        out_dt_define(macro, val)
+
+def write_vanilla_props(node):
+    # Writes macros for any and all properties defined in the
+    # "properties" section of the binding for the node.
+    #
+    # This does generate macros for special properties as well, like
+    # regs, etc. Just let that be rather than bothering to add
+    # never-ending amounts of special case code here to skip special
+    # properties. This function's macros can't conflict with
+    # write_special_props() macros, because they're in different
+    # namespaces. Special cases aren't special enough to break the rules.
+
+    macro2val = {}
+    for prop_name, prop in node.props.items():
+        macro = f"{node.z_path_id}_P_{str2ident(prop_name)}"
+        phandle = "phandle" in prop.type
+        val = prop2value(prop)
+        if val is not None:
+            # DT_N_<node-id>_P_<prop-id>
+            macro2val[macro] = val
 
+        if prop.enum_index is not None:
+            # DT_N_<node-id>_P_<prop-id>_ENUM_IDX
+            macro2val[macro + "_ENUM_IDX"] = prop.enum_index
 
-def write_spi_dev(node):
-    # Writes SPI device GPIO chip select data if there is any
+        plen = prop_len(prop)
```

Comment:
Could maybe have an `array_prop()` function that returns true/false instead, and do `if "phandle" not in prop.type and array_prop(prop)`.

Don't have to figure out that `None` length = not array then.